streams/shared/src/main/scala/zio/stream/SubscriptionRef.scala

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2017-2020 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.stream
+
+import zio.RefM
+
+/**
+ * A `SubscriptionRef[A]` contains a `RefM[A]` and a `Stream` that will emit
+ * every change to the `RefM`.
+ */
+final case class SubscriptionRef[A](ref: RefM[A], changes: Stream[Nothing, A])
```

Comment:
Should the constructor be private so we can guarantee it is created in the appropriate way?

Reply:
Good idea.

Reply:
Might as well also make it a class so it doesn't have `copy`.

Reply:
Already done. I'm moving the constructors there as well and adding one that lets you make it in the context of a managed.